File: 2125-number-of-laser-beams-in-a-bank/2125-number-of-laser-beams-in-a-bank.java
class Solution {
    public int numberOfBeams(String[] bank) {
        int res=0,temp=0;
        for(String s:bank){
            int cnt=0;
            for(char c:s.toCharArray()){
                if(c=='1')cnt++;
            }
            if(cnt>0){
                res+=(temp*cnt);
                temp=cnt;
            }
        }
        return res;
    }
}

File: 2125-number-of-laser-beams-in-a-bank/2125-number-of-laser-beams-in-a-bank.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        vector<int>temp;
        int ans=0;
        for(int i=0;i<bank.size();i++){
            int cnt=0;
            for(int j=0;j<bank[0].size();j++){
                if(bank[i][j]=='1')cnt++;
            }
            temp.push_back(cnt);
        }
        int res=0,t=0;
        for(int i=0;i<temp.size();i++){
            if(temp[i]>0){
               res+=t*temp[i];
                t=temp[i];
            }
        }
       return res;
    }
};